[inas0ng] iP

inas0ng opened the pull request:
# Duke
> "Clutter is postponed decisions." - Barbara Hemphill ([source](https://www.linkedin.com/pulse/clutter-everywhere-wake-up-call-america-barbara-hemphill))

Clean up clutter and start being organised with Duke. Keep a list of things you need to do! It's
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://github.com/inas0ng/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 🤩 

And it is **FREE**!

Features:
- [x] Managing tasks
- [x] Managing deadlines and events
- [ ] Reminders (coming soon)
---
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
updated Duke.java with Level 1. Greet, Echo, Exit

Commit:
edited Duke.java for Level 1

Commit:
updated Duke.java with Level 2. Add, List

Commit:
updated Duke.java and added Task.java for Level 3. Mark as Done

Commit:
updated Duke.java and added Deadline.java, Event.java and Todo.java for Level 4. ToDos, Events, Deadlines

Commit:
updated EXPECTED.TXT and input.txt for A-TextUiTesting

Commit:
updated Duke.java, EXPECTED.TXT and input.txt, added DukeException.java for Level 5. Handle Errors

Commit:
updated Duke.java, EXPECTED.TXT, input.txt for Level 6. Delete

Commit:
Add Storage.java and TaskList.java for Level 7

Commit:
Update Deadline.java, Duke.jave, Event.java for Level 8.

Commit:
Merge branch 'branch-Level-7'

Commit:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

Resolution: keep latest (Level 8)

Commit:
Add Parser.java and Ui.java for A-MoreOOP

Commit:
Package files in duke for A-Packages

Commit:
Add Gradle support for A-Gradle

Commit:
Add DeadlineTest.java and TaskListTest.java for A-JUnit

Commit:
Update build.gradle for A-Jar

Commit:
Add JavaDoc comments for A-JavaDoc

Commit:
Tweak code to comply with coding standard for A-CodingStandard

Commit:
Update Duke.java and TaskList.java for Level 9

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'branch-Level-9'

Commit:
Update files for A-CheckStyle

Commit:
Merge branch 'branch-A-CheckStyle'

Commit:
Add a GUI to Duke for Level 10

Commit:
Merge branch 'branch-Level-10'

Yaladah commented on src/main/java/duke/Duke.java:
```diff
@@ -44,6 +44,198 @@ public void saveTaskList(List<Task> taskList) throws IOException {
+            break;
+        case "deadline":
+            try {
+                String deadlineStr = parser.getDeadline(userInput);
+                String[] split = deadlineStr.split(" ");
+                LocalDate deadlineDate = LocalDate.parse(split[0]);
+                Optional<LocalTime> deadlineTime;
+                if (split.length > 1) {
+                    // Time is specified by user
+                    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("HHmm");
+                    deadlineTime = Optional.of(LocalTime.parse(split[1], formatter));
+                } else {
+                    deadlineTime = Optional.empty();
+                }
+                Deadline newDeadline = new Deadline(parser.getDeadlineDesc(userInput),
+                        parser.getDeadline(userInput));
+                tasks = tasks.add(newDeadline);
+                response.append(ui.lineTop());
+                response.append("i've added this task with deadline:\n" + newDeadline + "\n");
+                if (tasks.size() == 1) {
+                    response.append("you have 1 task in the list");
+                } else {
+                    response.append("you have " + tasks.size() + " tasks in the list");
+                }
+                response.append(ui.lineBottom());
```
Very clear and easy to understand code :+1:

Yaladah reviewed (commented):
LGTM!

arkobrat commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,436 @@
+     * @param userInput User input.
+     * @return Respnose from Duke.
+     */
+    public String getResponse(String userInput) {
```
Instead to aggregating all event handles (both parsing and UI response) in one method, would it not be better to add separate methods to separately handle separate commands?

gwynethguo commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,120 @@
+     * @param newTask The new task that replaces the existing task.
+     * @return TaskList with replaced task.
+     */
+    public TaskList set(int index, Task newTask) {
+        TaskList newList = new TaskList(tasks);
+        newList.tasks.set(index, newTask);
+        return newList;
+    }
+
+    /**
+     * Returns the Task at the index.
+     *
+     * @param index The index of the task to be retrieved.
+     * @return Task at the index.
+     */
+    public Task get(int index) {
+        return this.tasks.get(index);
+    }
+
+    /**
+     * Returns size of TaskList.
+     * @return Size of TaskList.
+     */
+    public int size() {
+        return this.tasks.size();
+    }
```
Perhaps a more intuitive variable name? (e.g. setTask, getTask, getSize)

gwynethguo commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,90 @@
+                    }
+                    break;
+                case "E":
+                    if (taskDetails[1].equals("1")) {
+                        tasks.add(new Event(taskDetails[2], true, taskDetails[3], taskDetails[4]));
+                    } else {
+                        tasks.add(new Event(taskDetails[2], false, taskDetails[3], taskDetails[4]));
+                    }
+                    break;
```
Perhaps can add line break here?

gwynethguo reviewed (commented):
Overall, your code aligns with the Java coding standard given in this module :)
Keep up the good work! :D

Commit:
Update build.gradle

.jar file generated uses CLI as it uses duke.Duke as the main class.

Let's change the build.gradle application main class to duke.Launcher
to use the GUI instead.

Commit:
Update Storage.java for A-Assertions

Storage stores the status of a task, marked or unmarked, with "1" and "0"
respectively.

Let's add assertions to ensure that upon load up, task status is only
represented by these values.

Commit:
Updated files for A-CodeQuality

Duke class: Minimise scope of variables, resolve warnings
MainWindow and Ui class: Remove "---" divider for the GUI

Variable index is defined without an initialisation in Duke.java.
StringBuilder is giving warnings due to the use of + operator.

Let's
* define int index in the least possible scope in Duke.java
* resolve StringBuilder warnings by using the append method instead
* delete dead code in Ui.java

inas0ng committed:
Merge pull request #1 from inas0ng/branch-A-Assertions

Update Storage.java for A-Assertions

Commit:
Merge branch 'master' into branch-A-CodeQuality

inas0ng committed:
Merge pull request #2 from inas0ng/branch-A-CodeQuality

Updated files for A-CodeQuality

Commit:
Merge branch 'master' of https://github.com/inas0ng/ip

inas0ng committed:
Create gradle.yml

inas0ng committed:
Update gradle.yml

Commit:
Update Launcher.java and MainWindow.java for checkstyle

Commit:
Update Duke.java for C-FriendlierSyntax

Commands available are todo, deadline, event, find, list, etc.

Let's add friendlier syntax options to save time typing longer commands!
* todo: t
* deadline: d
* event: e
* find: f
* list: l

Commit:
Update DialogBox.fxml to display full text in the GUI

Commit:
Add Ui.png as a representative screenshot of the product

inas0ng committed:
Update README.md

Commit:
Update Duke.java to prevent printing additional text

inas0ng committed:
Update README.md

inas0ng committed:
Update README.md
